discovery_engine_core/core/src/engine.rs

```diff
@@ -602,10 +602,10 @@ async fn update_stacks<'a>(
     }
 
     // only return an error if all stacks failed
-    if errors.len() < stacks.len() {
-        Ok(())
-    } else {
+    if !errors.is_empty() && errors.len() >= stacks.len() {
```

Comment:
The implication here is that we return an error if all stacks returned an error, right? Without having looked at the rest of the code, it's not possible for a stack to return more than one error, right?

Reply:
is not possible, as soon as we have an error we log it, add it and we break out of the loop.

Reply:
The `>` part in the check is to avoid to leave unexpected cases out more then something that is really needed with the current code.